test/tracer/rev-test-tracer.py

```diff
@@ -33,22 +32,22 @@
 # Define the simulation components
 # Instantiate all the CPUs
 for i in range(0, sim_nodes):
-    print("Building " + str(i))
-    comp_cpu = sst.Component("cpu" + str(i), "revcpu.RevCPU")
-    comp_cpu.addParams({
-          "verbose": 5,                       # Verbosity
-          "numCores": 1,                      # Number of cores
-          "clock": "1.0GHz",                  # Clock
-          "memSize": 1024*1024*1024,          # Memory size in bytes
-          "machine": f"[CORES:{arch}]",       # Core:Config; RV32I for all
-          "startAddr": "[CORES:0x00000000]",  # Starting address for core 0
-          "memCost": "[0:1:10]",              # Memory loads required 1-10 cycles
-          "program": rev_exe,                 # Target executable
-          "splash": 1,                        # Display the splash message
-          # "trcOp": "slli",                  # base command for tracing [default: slli]
-          # "trcLimit": 0,                    # Maximum number of trace lines [default: 0]
-          # "trcStartCycle": 0                # Starting trace cycle [default: 0]
-          })
+  print("Building "+ str(i))
+  comp_cpu = sst.Component("cpu" + str(i), "revcpu.RevCPU")
+  comp_cpu.addParams({
+	"verbose" : 5,                                # Verbosity
+        "numCores" : 1,                               # Number of cores
+	"clock" : "1.0GHz",                           # Clock
+        "memSize" : 1024*1024*1024,                   # Memory size in bytes
+        "machine" : f"[CORES:{arch}]",                  # Core:Config; RV32I for all
+        "startAddr" : "[CORES:0x00000000]",           # Starting address for core 0
+        "memCost" : "[0:1:10]",                       # Memory loads required 1-10 cycles
+        "program" : rev_exe,                          # Target executable
+        "splash" : 1,                                 # Display the splash message
+        # "trcOp": "slli",                            # base command for tracing [default: slli]
+        # "trcLimit": 0,                              # Maximum number of trace lines [default: 0]
+        # "trcStartCycle" : 0                         # Starting trace cycle [default: 0]
+        })
```

Comment:
Use the `flake8` tool to verify all `*.py` files are good. Type:
```
flake8  --max-line-length 131 file.py
```

`brew install flake8` or `pip install flake8` installs it.

Reply:
It appears that the `naked` attribute cannot be used with `varargs` function parameter lists. The compiler emits code which corrupts the stack.
```
0000000000010210 <rev_fast_printf>:
   10210:       00b13423                sd      a1,8(sp)
   10214:       00c13823                sd      a2,16(sp)
   10218:       00d13c23                sd      a3,24(sp)
   1021c:       02e13023                sd      a4,32(sp)
   10220:       02f13423                sd      a5,40(sp)
   10224:       03013823                sd      a6,48(sp)
   10228:       03113c23                sd      a7,56(sp)
   1022c:       000028b7                lui     a7,0x2
   10230:       3968889b                addw    a7,a7,918 # 2396 <exit-0xdd52>
   10234:       00000073                ecall
   10238:       00008067                ret
```
It stores `a1-a7` on the stack, corrupting its caller.

I'm trying to find a workaround.